boards/silabs/radio_boards/xg28_rb4401c/xg28_rb4401c.dts

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) 2025 Shontal Biton
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+/dts-v1/;
+#include <silabs/xg28/efr32zg28b322f1024im68.dtsi>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+#include <zephyr/dt-bindings/regulator/silabs_dcdc.h>
+#include "xg28_rb4401c-pinctrl.dtsi"
+
+/ {
+	model = "Silicon Labs BRD4401C";
+	compatible = "silabs,xg28_rb4401c", "silabs,efr32zg28";
+
+	chosen {
+		zephyr,console = &usart0;
+		zephyr,shell-uart = &usart0;
+		zephyr,uart-pipe = &usart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,display = &ls0xx_ls013b7dh03;
+	};
+
+	aliases {
+		led0 = &led0;
+		led1 = &led1;
+		sw0 = &button0;
+		sw1 = &button1;
+		watchdog0 = &wdog0;
+		dht0 = &si7021;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led0: led_0 {
+			gpios = <&gpiob 2 GPIO_ACTIVE_HIGH>;
+			label = "LED 0";
+		};
+
+		led1: led_1 {
+			gpios = <&gpiod 3 GPIO_ACTIVE_HIGH>;
+			label = "LED 1";
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		
+		button0: button_0 {
+			gpios = <&gpiob 1 GPIO_ACTIVE_LOW>;
+			label = "User Push Button 0";
+			zephyr,code = <INPUT_KEY_0>;
+		};
+
+		button1: button_1 {
+			gpios = <&gpiob 3 GPIO_ACTIVE_LOW>;
+			label = "User Push Button 1";
+			zephyr,code = <INPUT_KEY_1>;
+		};
+	};
+
+	sensor_disp_enable: sensor_disp_enable {
+		compatible = "regulator-fixed";
+		enable-gpios = <&gpioc 11 GPIO_ACTIVE_HIGH>;
+		regulator-name = "sensor_disp_enable";
+		status = "okay";
+	};
+};
+
+&cpu0 {
+	clock-frequency = <78000000>;
+};
+
+&pstate_em3 {
+	status = "disabled";
+};
+
+&hfxo {
+	status = "okay";
+	ctune = <106>;
+	precision = <50>;
+};
+
+&lfxo {
+	status = "okay";
+	ctune = <38>;
+	precision = <50>;
+};
+
+&hfrcodpll {
+	clock-frequency = <DT_FREQ_M(78)>;
+	clocks = <&hfxo>;
+	dpll-n = <3839>;
+	dpll-m = <1919>;
+	dpll-edge = "fall";
+	dpll-lock = "phase";
+	dpll-autorecover;
+};
+
+&em23grpaclk {
+	clocks = <&lfxo>;
+};
+
+&em4grpaclk {
+	clocks = <&lfxo>;
+};
+
+&sysrtcclk {
+	clocks = <&lfxo>;
+};
+
+&wdog0clk {
+	clocks = <&lfxo>;
+};
+
+&wdog1clk {
+	clocks = <&lfxo>;
+};
+
+&usart0 {
+	current-speed = <115200>;
+	pinctrl-0 = <&usart0_default>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&eusart1 {
```

Comment:
Please follow the [Devicetree Style Guidelines](https://docs.zephyrproject.org/latest/contribute/style/devicetree.html)
You can look at https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/silabs/radio_boards/xg23_rb4210a/xg23_rb4210a.dts for reference.